lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3257,6 +3257,48 @@ void ClangImporter::Implementation::lookupAllObjCMembers(
   }
 }
 
+bool ClangImporter::Implementation::loadNamedMembers(
+    const Decl *D, DeclName N, uint64_t contextData,
+    TinyPtrVector<ValueDecl *> &Members) {
+
+  auto *DC = dyn_cast<DeclContext>(D);
+  if (!DC)
+    return true;
+
+  auto *CD = D->getClangDecl();
+  if (!CD)
+    return true;
+
+  auto CMO = getClangSubmoduleForDecl(CD);
+  if (!CMO || !*CMO)
+    return true;
+
+  auto table = findLookupTable(*CMO);
+  if (!table)
+    return true;
+
+  clang::ASTContext &clangCtx = getClangASTContext();
+
+  if (auto *CPD = dyn_cast<clang::ObjCProtocolDecl>(CD)) {
+    for (auto entry : table->lookup(SerializedSwiftName(N.getBaseName()), CPD)) {
+      if (!entry.is<clang::NamedDecl *>()) continue;
+      auto member = entry.get<clang::NamedDecl *>();
+      if (!isVisibleClangEntry(clangCtx, member)) continue;
+      SmallVector<Decl*, 4> tmp;
+      insertMembersAndAlternates(member, tmp);
+      for (auto *TD : tmp) {
+        if (auto *V = dyn_cast<ValueDecl>(TD)) {
+          Members.push_back(V);
```

Comment:
You need to be filtering these by name. Remember that a single Clang declaration may get imported under multiple Swift names.

Reply:
I'm not sure I follow. I'm looking up _by_ swift name, and I'm doing insertMembersAndAlternates to get the extra swift names it might be imported-as. Which names should I be filtering out? The alternates?

Reply:
No, the "alternates" are things like getters and setters for a property, or the original methods for Objective-C subscripts. I guess those do need to be filtered, but I was referring more to things like this:

```objc
- (void)foo NS_SWIFT_NAME(bar);
```

This will show up in the lookup tables for both `foo` and `bar`, and will import the method under both names, but we only want to return the one with the matching name. You can take a look at the way top-level lookup does this in `ClangImporter::Implementation::lookupValue`.